### PR TITLE
Remove extra FrameContext class

### LINKED ISSUE #837: FrameContext is an unneeded class
Everywhere that uses it, uses it to resolves `IFrameControl` only, which `Frame` could happily directly implement so it doesn't seem to add anything.

### PR DESCRIPTION
**Pull request by benaadams**
It doesn't particularly do anything

Resolves #837

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Http
 {
     class FrameResponseStream : Stream
     {
-        private FrameContext _context;
+        private IFrameControl _context;
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs]: Please rename it to `_frameControl`.

Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs]: Done

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs]: Forgot to push? :smile:

File: src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs [hunk 2]
```diff
@@ -134,14 +134,14 @@ public void Abort()
             }
         }
 
-        public void Initialize(FrameContext context)
+        public void Initialize(IFrameControl context)
         {
-            _context = context;
```
Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs]: We could do the same rename in MessageBody.cs and TestInput.cs (though the test stuff is less important to me).

**Comment by halter73** [comment]
Other than the finish of the context->frame rename, LGTM.

File: src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs (hunk 2 above)
Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameResponseStream.cs]: MessageBody is dependent on a few more bits, SocketInput, RejectRequest, TakeMessageHeaders. Could add to IFrameControl?

**Comment by benaadams** [comment]
I think I've got all the renames now?

**Comment by cesarbs** [comment]
:ship:

**Comment by halter73** [comment]
:shipit:

**Pull request merged**